docs/how-to/modify-kernel-boot-parameters.rst

```diff
@@ -0,0 +1,140 @@
+How to modify kernel boot parameters
+=======================================
+
+The real-time performance-tuning features compiled into the kernel can be
+controlled by various means. The approach used here is a common one:
+passing command line parameters via a bootloader.
+
+The Linux kernel supports a wide range of parameters, but only a few of them are relevant to real-time tuning.
+:doc:`/reference/kernel-boot-parameters` is good a reference.
+
+Depending on the underlying boot mechanism, there are multiple ways to pass command line parameters to the kernel.
+
+For Raspberry Pi, jump to :ref:`raspberry-pi`.
+
+Ubuntu Core uses GRUB but provides a different interface for 
+`modifying kernel options <https://ubuntu.com/core/docs/modify-kernel-options>`_.
+
+.. Replace the Ubuntu Core link to the internal how-to (https://documentation.ubuntu.com/real-time/en/ubuntu-core-devel/how-to/uc-boot-parameters/), once released.
+
+Continue reading for systems that use the GNU GRUB bootloader.
+
+Modify kernel boot parameters via GRUB
+--------------------------------------
+
+You can temporarily or permanently configure parameters which GRUB will pass to
+Real-time Ubuntu.
+
+When first testing a real-time tuning feature, `configure it temporarily
+<#configure-grub-temporarily>`_. Temporary bootloader parameters are passed to
+the kernel on a single startup. To re-use them you'll need to re-enter them ---
+they don't persist across reboots.
+
+When you're satisfied that a parameter achieves what you want, you can make it
+permanent by `updating GRUB's configuration <#configure-grub-permanently>`_
+then rebooting.
+
+Configure GRUB temporarily
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Reboot, and interrupt GRUB's startup sequence at its bootloader menu.
+
+Highlight the entry --- usually **Ubuntu** --- that would boot Real-time
+Ubuntu.
+
+.. image:: grub-menu.png
+   :alt: GRUB menu
+   :width: 100%
+   :align: center
+
+Press **e** to edit its boot parameters.
+
+Append parameters to the line that starts with ``linux``:
+
+.. image:: grub-edit.png
+   :alt: GRUB edit
+   :width: 100%
+   :align: center
+
+* Leave a space before each new parameter
+
+* Don't add space round ``=`` and other punctuations for each key-value
+
+* Don't add line breaks
```

Comment:
I don't think it's necessary to repeat these instructions three times. It would be better to add a `note` section before introducing the temporary GRUB configuration, stating that it is valid for the following configurations.

Reply:
I agree it's a bit repetitive and should be refactored somehow.

Also modifying kernel boot parameters isn't something just anyone would be doing; we should be able to expect a certain level of competency from the user doing this. And in that case the example (with at least 2 parameters) should be sufficient.

But I am OK to leave this out of this PR for now.

Reply:
On a related note, I tried comparing what other docs did for this.
Surprisingly none of them actually indicated the editing rules for these files.

For reference:
https://wiki.ubuntu.com/Kernel/KernelBootParameters#Permanently_Add_a_Kernel_Boot_Parameter
https://wiki.archlinux.org/title/kernel_parameters
https://wiki.archlinux.org/title/kernel_parameters#GRUB

Reply:
I think it is good to mention these directly when the user is making the changes. A note somewhere on top could be forgotten. Adding a section and referring to it is also not ideal because it breaks the flow for something as small as this. 

With that said, the text is mostly copied from cpu config doc to another one. I prefer making separate PRs to make improvements that are outside the scope of this PR: adding Pi config.

Reply:
@AnneCYH I wrote the above comment before seeing your replies.

Let's consider these in another PR.

> Surprisingly none of them actually indicated the editing rules for these files.

The Ubuntu wiki has the following, which is even more detailed:

> Press SPACE to add a blank space (after "splash") then carefully type in the kernel boot parameter that you need to add. (If you need to add multiple parameters separate them with SPACE but do not add spaces before or after any = signs or punctuation in the parameters themselves).